### PR TITLE
feat(notes): ship a Bases-friendly default episode note template (#160)

### LINKED ISSUE #160: FR: Add Obsidian Bases in default note template
Please consider adding Obsidian Bases in default note template, so that we can take advantage of the bases views, sorting, filtering etc.

As an example, these are the properties I am using in my template:

---
Podcast: "{{podcast}}"
URL: "{{url}}"
tags:
  - Podcast
Rating:
Status: Listened / Current / To Listen
Favourite:
Start Date:
Listened Date:
---

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary

Closes #160.

The episode note defaults shipped empty (`note.path = ""`, `note.template = ""`), which left the "Create episode note" command disabled on a fresh install and gave new users no structured metadata to build [Obsidian Bases](https://help.obsidian.md/bases) views on. This PR ships a sensible, Bases-friendly default for both:

- `note.path`: `PodNotes/{{podcast}}/{{title}}.md` (matches the existing `download.path` convention).
- `note.template`: YAML frontmatter with structured properties Bases can sort, filter, and group on, plus user-fillable tracking fields:

```
---
type: podcastEpisode
podcast: "{{podcastlink}}"
url: "{{url}}"
date: {{date:YYYY-MM-DD}}
tags:
  - podcastEpisode
status:
rating:
favorite: false
---
# {{title}}

![]({{artwork}})

[Resume in PodNotes]({{episodelink}})

{{description}}
```

`{{podcastlink}}` ties each episode to its podcast feed note (the #163 feature) so Bases / Dataview can roll episodes up under their show. The raw `{{title}}` lives in the body H1, where YAML rules do not apply, so awkward titles never break the frontmatter.

## What changed

- **`src/constants.ts`** - new `note.path` / `note.template` defaults (scoped to the `note` block only).
- **`src/TemplateEngine.ts`** - `NoteTemplateEngine` now sanitizes the URL tags (`url`, `stream`, `artwork`, `episodeurl`, `episodeartwork`, `feedurl`, `feedartwork`) with the existing `sanitizeUrlForTemplate` (strips `"` and `\`), mirroring `FeedNoteTemplateEngine`. This is lossless for well-formed URLs and keeps a quoted scalar like `url: "{{url}}"` valid. Each tag is guarded with `?? ""` so a corrupted `data.json` can't throw.
- **`src/settingsMigrations.ts`** - new `migrateNoteSettings`, which upgrades the legacy empty note to the new default **only when the whole note is the old default** (both path and template empty/absent). Any configured field is preserved verbatim, including a deliberately-empty field used to keep note creation disabled.
- **`src/main.ts`** - `loadSettings` applies `migrateNoteSettings`.
- **`scripts/provision-obsidian-e2e-vault.mjs`** - the e2e seed note block now matches `DEFAULT_SETTINGS` (keeps the seed/DEFAULT_SETTINGS drift test green and seeds the new default for verification).
- **`src/ui/settings/PodNotesSettingsTab.ts`** - the note-template placeholder now shows a Bases-friendly frontmatter hint instead of the old Dataview `::` style.
- **`docs/docs/templates.md`** - documents the new default, an example `.base` view, and the URL-tag sanitization.

## Compatibility / migration impact

Defaults only reach **new installs** and users who never configured note settings. Existing users are protected:

- The migration fires only when the stored note is exactly the old default (`{ path: "", template: "" }`). A user who customized either field keeps both fields untouched.
- This deliberately preserves the "custom path + empty template to keep note creation disabled" case (raised in review).
- No storage shape change; the migrated value is re-persisted on the next save (same pattern as the #183 `migrateDownloadPath`).

The URL-tag sanitization changes existing tag output only for malformed inputs (a `"` or `\` in a URL, which valid URLs never contain). For local-file episodes `{{url}}` is a wikilink; a normal filename is unaffected, and a filename containing a literal `"` would have its quote stripped (rare; valid YAML is preferred over an exact link in that edge case).

## Verification

Gates (Node 22), all green:

```
npm run lint
npm run format:check
npm run typecheck
npm run build
npm run test        # 466 passing
npm run docs:build  # mkdocs --strict
```

Real-Obsidian verification in an isolated worktree vault (`npm run obsidian:e2e`): drove the actual `create-podcast-note` command for several episodes, including a hostile title/URL (`Why "AI": a deep dive: part 2`, URL with an embedded quote and backslash) and a no-date episode. Obsidian's properties parser produced clean frontmatter in every case:

- hostile: `url: "https://example.com/ep?x=qc"`, `podcast: "[[PodNotes/Podcasts/My Show|My Show]]"`, `tags: [podcastEpisode]`, `status: null`, `rating: null`, `favorite: false`; the raw title sits only in the body H1.
- no-date: `date: null`.

## Review

Ran a multi-dimension self-review plus two adversarial reviewers. Key fixes that came out of it: the whole-note migration gate (so an intentional empty field is never re-enabled), correcting the example `.base` to the documented `groupBy` form (the canonical Bases schema has no `sort` view key), and qualifying the "valid YAML" claim for hand-edited feed-note paths that contain YAML-hostile characters.

### CODE REVIEW

File: src/TemplateEngine.ts [hunk 1]
```diff
@@ -120,8 +120,15 @@ export function NoteTemplateEngine(template: string, episode: Episode) {
...
-	addTag("url", episode.url);
+	// URL tags are sanitized (quote/backslash stripped) so they stay valid inside a
+	// quoted YAML frontmatter scalar — the Bases-friendly default note template
+	// quotes {{url}}. The strip is lossless for well-formed URLs (a valid URL never
+	// contains a raw double-quote or backslash) and mirrors FeedNoteTemplateEngine.
+	// The `?? ""` keeps a corrupted/hand-edited data.json (where a typed-string URL
+	// is actually null/undefined) from throwing in sanitizeUrlForTemplate and
+	// aborting note creation — same guard the other URL tags below use. See #160.
+	addTag("stream", sanitizeUrlForTemplate(episode.streamUrl ?? ""));
+	addTag("url", sanitizeUrlForTemplate(episode.url ?? ""));
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/TemplateEngine.ts]: Good catch, thanks. You are right that episode.url is a wikilink for local-file episodes (generateMarkdownLink), so globally stripping characters would corrupt the link and would also change {{url}} for body uses.

I reverted the URL-tag sanitization in NoteTemplateEngine back to verbatim rendering (commit 1d59e3c). The default frontmatter still renders valid YAML for the common case: feed episode URLs are well-formed, the {{podcastlink}} name is already sanitized, and ordinary local-file links contain no quotes. A URL or file name containing a literal quote is the one case that would need adjusting, which is now noted in the docs. sanitizeUrlForTemplate stays in use by FeedNoteTemplateEngine, where {{url}} is always a real website URL. Added a test that {{url}}/{{episodeurl}} pass through verbatim so local-file wikilinks are preserved.

**Comment by chhoumann** [comment]
@codex review

Rebased onto master (resolved the src/constants.ts overlap with #114, keeping both the new episodeListLimit setting and the note.template default) and addressed your P2 by reverting the URL-tag sanitization to verbatim rendering in NoteTemplateEngine (commit 1d59e3c). Gates are green. Could you take another look?

**Comment by chhoumann** [comment]
@codex review

File: src/constants.ts [hunk 2]
```diff
@@ -82,8 +82,38 @@ export const DEFAULT_SETTINGS: IPodNotesSettings = {
...
+		// need adjusting. The raw {{title}} (which may contain quotes/colons) lives
+		// in the note body as the H1, where YAML rules don't apply. status/rating/
+		// favorite are left for the user to fill and give Bases columns to sort and
+		// filter on. {{podcastlink}} ties each episode to its feed note (#163). See
+		// issue #160.
+		template:
+			"---\n" +
+			"type: podcastEpisode\n" +
+			'podcast: "{{podcastlink}}"\n' +
+			'url: "{{url}}"\n' +
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/constants.ts]: Agreed, and thanks for catching the other side of this. Rather than escape, I moved {{url}} out of the quoted frontmatter scalar entirely (commit b698643). The default frontmatter now contains only values that can never carry a YAML-hostile character (type, the sanitized {{podcastlink}}, an ISO/empty date, tags, status/rating/favorite), so a local-file link with a quote can't break the parse. {{url}} now renders in the body, where it is safe for both feed URLs and vault links. Feed-only users can add url: "{{url}}" back, which I noted in the docs. Added a test that exercises a quote-bearing local-file url to prove the frontmatter stays parseable.

**Comment by chhoumann** [comment]
@codex review
